src/Symfony/Component/Ldap/Security/LdapUserProvider.php

```diff
@@ -73,17 +72,10 @@ public function loadUserByUsername(string $username): UserInterface
 
     public function loadUserByIdentifier(string $identifier): UserInterface
     {
-        try {
-            $this->ldap->bind($this->searchDn, $this->searchPassword);
-            $identifier = $this->ldap->escape($identifier, '', LdapInterface::ESCAPE_FILTER);
-            $query = str_replace(['{username}', '{user_identifier}'], $identifier, $this->defaultSearch);
-            $search = $this->ldap->query($this->baseDn, $query, ['filter' => 0 == \count($this->extraFields) ? '*' : $this->extraFields]);
-        } catch (ConnectionException $e) {
-            $e = new UserNotFoundException(sprintf('User "%s" not found.', $identifier), 0, $e);
-            $e->setUserIdentifier($identifier);
-
-            throw $e;
-        }
+        $this->ldap->bind($this->searchDn, $this->searchPassword);
```

Comment:
We could opt to catch InvalidCredentialsException here in order to throw something more-specific that would indicate it is the search bind credentials (used during enumeration) that are invalid, as opposed to the credentials that the user has provided.

Reply:
Same in `CheckLdapCredentialsListener.php` line 77

Point being that throwing the `InvalidCredentialsException` can be ambiguous in either meaning "user-provided credentials are invalid" or "configured search credentials are invalid"

Reply:
https://github.com/symfony/symfony/pull/44898/commits/7ab0611440472173a604b35e1eaaa7791a0569f7 introduces an `InvalidSearchCredentialsException` which is thrown when binding fails when using the configured search credentials.